seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -375,44 +378,72 @@ def test_validate_anvil_vcf(self, mock_file_iter, mock_file_exist, mock_utils_lo
         self.assertEqual(response.status_code, 400)
         self.assertEqual(response.json()['error'],
                          'Invalid VCF file format - file path must end with .vcf or .vcf.gz or .vcf.bgz')
 
+        # Test bad sharded VCF file path
+        mock_subprocess.return_value.stdout = [b'File not found']
+        response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_SHARDED_DATA_PATH))
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()['error'], 'Loading VCF file list for gs://test_bucket/test_path-*.vcf.gz failed with error '
+                                                   'Run command failed: File not found')
+        mock_subprocess.assert_called_with('gsutil -q ls gs://test_bucket/test_path-*.vcf.gz', stdout=-1, stderr=-2, shell=True)
+        mock_file_logger.info.assert_called_with('==> gsutil -q ls gs://test_bucket/test_path-*.vcf.gz', self.manager_user)
+        mock_api_logger.error.assert_called_with(
+            'Loading VCF file list for gs://test_bucket/test_path-*.vcf.gz failed with error Run command failed: File not found',
+            self.manager_user
+        )
+
         # test no header line
-        mock_file_exist.return_value = True
-        mock_file_iter.return_value = BASIC_META + DATA_LINES
+        mock_subprocess.reset_mock()
+        mock_subprocess.return_value.wait.return_value = 0
+        mock_subprocess.return_value.stdout = [s.encode('utf-8') for s in BASIC_META + DATA_LINES]
         response = self.client.post(url, content_type='application/json', data=json.dumps(REQUEST_BODY_GZ_DATA_PATH))
         self.assertEqual(response.status_code, 400)
-        self.assertListEqual(response.json()['errors'], [
-            'No header found in the VCF file.'
+        self.assertListEqual(response.json()['errors'], ['No header found in the VCF file.'])
+        mock_subprocess.assert_has_calls([
+            mock.call('gsutil ls gs://test_bucket/test_path.vcf.gz', stdout=-1, stderr=-2, shell=True),
+            mock.call().wait(),
+            mock.call('gsutil cat -r 0-65536 gs://test_bucket/test_path.vcf.gz | gunzip -c -q - ',
+                      stdout=-1, stderr=-2, shell=True),
+        ])
+        mock_file_logger.info.assert_has_calls([
+            mock.call('==> gsutil ls gs://test_bucket/test_path.vcf.gz', self.manager_user),
+            mock.call('==> gsutil cat -r 0-65536 gs://test_bucket/test_path.vcf.gz | gunzip -c -q - ', None),
         ])
-        mock_file_iter.assert_called_with('gs://test_bucket/test_path.vcf.gz', byte_range=(0, 65536))
-        mock_file_exist.assert_called_with('gs://test_bucket/test_path.vcf.gz', user=self.manager_user)
 
         # test header errors
-        mock_file_iter.return_value = BASIC_META + BAD_INFO_META + BAD_FORMAT_META + BAD_HEADER_LINE + DATA_LINES
+        mock_subprocess.return_value.stdout = [s.encode('utf-8') for s in
```

Comment:
wouldn't it be better to update the fixture data to be encoded, instead of calling encode every time?